### ISSUE TITLE
Power Platform Build Tools V2 exporting as managed has default postfix of _managed

### ISSUE THREAD

**Issue by m3ngi3** [MicrosoftDocs/power-platform#1488]
Using the new @2 version (https://powerapps.microsoft.com/en-us/blog/new-version-of-power-platform-build-tools-and-a-feature-packed-update-of-power-platform-cli/) it seems the export command postfixes _managed behind the .zip file...

`
      - task: PowerPlatformExportSolution@2  #Export Managed Solution to machine as .zip file locally
        displayName: 'Solutions - ManSol export zip locally'
        inputs:
          authenticationType: 'PowerPlatformSPN'
          PowerPlatformSPN: ${{ parameters.connectionString }}
          SolutionName: '${{ parameters.SolutionInternalName }}'
          SolutionOutputFile: '$(GitDirectoryExportedSolutions)\${{ parameters.SolutionInternalName }}M.zip'
          Managed: true
          AsyncOperation: true
          MaxAsyncWaitTime: '60'
          ExportAutoNumberingSettings: true
`

Results in 
![image](https://user-images.githubusercontent.com/47530580/192953238-a0bbf84d-cdcf-4c0d-af89-5021df1248c1.png)

Is that intentional? In previous version this was not so --> I recommend to reflect this in the documentation?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: e2d7f63d-dcde-f057-555a-3fffce0f229c
* Version Independent ID: e4bba5f7-ae53-260f-269b-c8ad002ef696
* Content: [Build tool tasks - Power Platform](https://learn.microsoft.com/en-us/power-platform/alm/devops-build-tool-tasks)
* Content Source: [power-platform/alm/devops-build-tool-tasks.md](https://github.com/MicrosoftDocs/power-platform/blob/main/power-platform/alm/devops-build-tool-tasks.md)
* Service: **power-platform**
* Sub-service: **alm**
* GitHub Login: @mikkelsen2000
* Microsoft Alias: **pemikkel**

**Comment by bishalgoswami** [comment]
Thanks for your comment, @m3ngi3. This issue has been assigned to the doc author for review.

**Comment by kkanakas** [comment]
@m3ngi3 that is not the case the only time it does it on my setup is when I have the export setup as this 
<img width="1503" alt="image" src="https://user-images.githubusercontent.com/22942069/193888527-8b209e5a-e141-4ffd-8600-97c6d5fd07f8.png">

do you have something similar ?

**Comment by m3ngi3** [comment]
No --> see code when using updated Version 2 of Power Platform Build Tools:
`
      - task: PowerPlatformExportSolution@2  #Export Managed Solution to machine as .zip file locally
        displayName: 'Solutions - ManSol export zip locally'
        inputs:
          authenticationType: 'PowerPlatformSPN'
          PowerPlatformSPN: ${{ parameters.connectionString }}
          SolutionName: '${{ parameters.SolutionInternalName }}'
          SolutionOutputFile: '$(GitDirectoryExportedSolutions)\${{ parameters.SolutionInternalName }}M.zip'
          Managed: true
          AsyncOperation: true
          MaxAsyncWaitTime: '60'
          ExportAutoNumberingSettings: true
`
When the Task after this Task is:

`
      - task: PowerPlatformUnpackSolution@2 #Unpack Managed Solution .zip file locally
        displayName: 'Solutions - ManSol zip unpack locally'
        inputs:
          SolutionInputFile: '$(GitDirectoryExportedSolutions)\${{ parameters.SolutionInternalName }}M.zip'
          SolutionTargetFolder: '$(GitDirectoryExportedSolutions)\${{ parameters.SolutionInternalName }}Managed' 
          SolutionType: Managed 
`

the second task resulted in:
![image](https://user-images.githubusercontent.com/47530580/194001460-8089e77f-601e-4dd5-a915-f2de06d7ceab.png)


please note the **SolutionOutputFile** of first Task matching the **SolutionInputFile** in second Task exactly...

When I change the second Task to:
`
      - task: PowerPlatformUnpackSolution@2 #Unpack Managed Solution .zip file locally
        displayName: 'Solutions - ManSol zip unpack locally'
        inputs:
          SolutionInputFile: '$(GitDirectoryExportedSolutions)\${{ parameters.SolutionInternalName }}M_managed.zip'
          SolutionTargetFolder: '$(GitDirectoryExportedSolutions)\${{ parameters.SolutionInternalName }}Managed' 
          SolutionType: Managed 
`
it completed successfully. 

So where does the _managed postfix come from?

**Comment by kkanakas** [comment]
it could be from the variable that you are setting for the SolutionInternalName, where are you setting that ?

**Comment by m3ngi3** [comment]
Nope... that comes from a template parameter that comes from a Library Variable 

Pipeline stage = 
- stage: 'buildinsbadev01'
  displayName: 'Build from insbadev01'
  jobs:
    - job: 'Build01'
      displayName: '$(PowerPlatformSolution02) build'
      steps:
      - template: 'Templates/BuildSolution.yml'
        parameters:
          connectionString: '$(EnvironmentServiceConnection)'
          environment: 'insbadev01'
          gitBrancheObject: GitRepoBuild
          SolutionInternalName: $(PowerPlatformSolution02)        
          SolutionVersionPrefix: '1.0'
          gitUserEmail: '$(gitUserEmailPipeline)'

Variable example = 
![image](https://user-images.githubusercontent.com/47530580/194230332-8bcdaff1-9a36-4745-994c-b1a1025ebe8c.png)

--> it should match the exact internal name of the Power Platform otherwise the export action would fail because the solution is not named with _managed at the back.

So you cannot reproduce using this exact same version of the Power Platform Build Tool Task in a DevOps Pipeline?
I have reproduced it in multiple tenants, multiple devops projects and multiple git repo's... 
I even have it reproduced when I export a solution manually:
![image](https://user-images.githubusercontent.com/47530580/194230705-60de2cb8-2685-46a8-8c82-1582afffbb1b.png)

When you export it you get the exact same give output path without _managed patched behind it??

**Comment by m3ngi3** [comment]
Hi @kkanakas ,
I see you were able to reproduce?? 😉
https://powerapps.microsoft.com/en-us/blog/power-platform-developer-tools-monthly-release-september-refresh/ 
![image](https://user-images.githubusercontent.com/47530580/197465060-b4e96ebf-47db-4038-8577-3cd362037ad0.png)

In my Export tasks you will see that I have defined the **SolutionOutputFile** property so why does it still add _managed behind the output file?

**Comment by kkanakas** [comment]
Yes, it is a new feature. Can you do me a favor and open a case with support and provide the logs and the steps that you are doing how you are running into this situation.


Kartik (He/Him)
I work in Seattle Time (Pacific Time).My working hours may not be your working hours. Please do not feel obligated to respond outside of your normal work hours.

**Comment by m3ngi3** [comment]
Hi @kkanakas,
It seems I am not able to reproduce anymore... when the output file path is specified it seems to respect it now and not place _managed behind it anymore. That is great!

But as you can see above in one of my posts that unpacking the file using **the exact same output file path** --> the file could not be found because of the _managed postfix. These kind of bugs are a real headache because of Azure DevOps Pipeline automations that suddenly stop working. Please be aware when releasing these kind of updates 👍.
